From c79718fb69d9a354a676de07b5ea45bdb5f6b53d Mon Sep 17 00:00:00 2001
From: Andrew Wang <andrew.wang@cloudera.com>
Date: Thu, 25 Sep 2014 22:06:23 -0700
Subject: [PATCH 196/596] CLOUDERA-BUILD. Fix up HDFS-6826 because of
 HDFS-7077 changes to ClientProtocol.

(cherry picked from commit e0bd2e4d2b3505eff13f2e91934dc022bba5422f)
---
 .../AuthorizationProviderProxyClientProtocol.java  |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/AuthorizationProviderProxyClientProtocol.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/AuthorizationProviderProxyClientProtocol.java
index 9350714..61bcd43 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/AuthorizationProviderProxyClientProtocol.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/AuthorizationProviderProxyClientProtocol.java
@@ -17,7 +17,7 @@
  */
 package org.apache.hadoop.hdfs.server.namenode;
 
-import org.apache.hadoop.crypto.CipherSuite;
+import org.apache.hadoop.crypto.CryptoProtocolVersion;
 import org.apache.hadoop.fs.BatchedRemoteIterator;
 import org.apache.hadoop.fs.CacheFlag;
 import org.apache.hadoop.fs.ContentSummary;
@@ -96,7 +96,7 @@ public FsServerDefaults getServerDefaults() throws IOException {
   @Override
   public HdfsFileStatus create(String src, FsPermission masked,
       String clientName, EnumSetWritable<CreateFlag> flag, boolean createParent,
-      short replication, long blockSize, List<CipherSuite> cipherSuites)
+      short replication, long blockSize, CryptoProtocolVersion[] supportedVersions)
       throws AccessControlException, AlreadyBeingCreatedException,
              DSQuotaExceededException, FileAlreadyExistsException,
              FileNotFoundException, NSQuotaExceededException,
@@ -106,7 +106,7 @@ public HdfsFileStatus create(String src, FsPermission masked,
     try {
       AuthorizationProvider.beginClientOp();
       return server.create(src, masked, clientName, flag, createParent,
-          replication, blockSize, cipherSuites);
+          replication, blockSize, supportedVersions);
     } finally {
       AuthorizationProvider.endClientOp();
     }
-- 
1.7.9.5

